src/Storageless/Session/DefaultSessionData.php

```diff
@@ -133,24 +145,36 @@ public function hasChanged() : bool
      */
     public function isEmpty() : bool
     {
-        return empty($this->data);
+        return ! count($this->data);
     }
 
     /**
      * {@inheritDoc}
      */
-    public function jsonSerialize()
+    public function jsonSerialize() : object
     {
-        return $this->data;
+        return (object) $this->data;
     }
 
     /**
-     * @param int|bool|string|float|array|object|\JsonSerializable $value
+     * @param int|bool|string|float|mixed[]|object|\JsonSerializable|null $value
      *
-     * @return int|bool|string|float|array
+     * @return int|bool|string|float|mixed[]
      */
     private static function convertValueToScalar($value)
     {
-        return json_decode(json_encode($value, \JSON_PRESERVE_ZERO_FRACTION), true);
+        $jsonScalar = json_encode($value, JSON_PRESERVE_ZERO_FRACTION);
+
+        if (! is_string($jsonScalar)) {
+            // @TODO use PHP 7.3 and JSON_THROW_ON_ERROR instead? https://wiki.php.net/rfc/json_throw_on_error
```

Comment:
This is to be discussed: IMO, we can go with the current solution, and I'd open a new issue to upgrade once 7.3 is released and available in CI.

Reply:
Leaving this as-is - upgrade to 7.3 may be applied later, once the tooling also understands `JSON_THROW_ON_ERROR`